misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/MatchResourcewithOperationId.java

```diff
@@ -0,0 +1,534 @@
+ */
+public class MatchResourcewithOperationId {
+    /**
+     * Filter all the operations according to the given filters.
+     * @param openApi       OpenApi Object
+     * @param filters       Filter Object
+     * @return              List of OpenApiPathSummary
+     */
+    public static List<OpenAPIPathSummary> filterOpenapi(OpenAPI openApi,
+                                                         Filters filters) {
+
+        boolean tagFilteringEnabled = filters.getTag().size() > 0;
+        boolean operationFilteringEnabled = filters.getOperation().size() > 0;
+        boolean excludeTagsFilteringEnabled = filters.getExcludeTag().size() > 0;
+        boolean excludeOperationFilteringEnable = filters.getExcludeOperation().size() > 0;
+
+        List<OpenAPIPathSummary> openAPIPathSummaries = MatchResourcewithOperationId.summarizeOpenAPI(openApi);
+        // Check based on the method and path filters
+        Iterator<OpenAPIPathSummary> openAPIIter = openAPIPathSummaries.iterator();
+        while (openAPIIter.hasNext()) {
+            OpenAPIPathSummary openAPIPathSummary = openAPIIter.next();
+            // If operation filtering available proceed.
+            // Else if proceed to check exclude operation filter is enable
+            // Else check tag filtering or excludeTag filtering enable.
+            if (operationFilteringEnabled) {
+                // If tag filtering available validate only the filtered operations grouped by given tags.
+                // Else if exclude tag filtering available validate only the operations that are not include
+                // exclude Tags.
+                // Else proceed only to validate filtered operations.
+                if (tagFilteringEnabled) {
+                    Iterator<Map.Entry<String, Operation>> operations =
+                            openAPIPathSummary.getOperations().entrySet().iterator();
+                    while (operations.hasNext()) {
+                        Map.Entry<String, Operation> operationMap = operations.next();
+//                        Check operationId is null scenario.
+                        if ((filters.getOperation().contains(operationMap.getValue().getOperationId())) ||
+                                operationMap.getValue().getOperationId() == null) {
+//                                    Check tag is available if it is null then remove other wise else-if not include
+//                                    tag then remove operations.
+                            if (operationMap.getValue().getTags() == null) {
+                                operations.remove();
+                            } else if (Collections.disjoint(filters.getTag(), operationMap.getValue().getTags())) {
+//                                        Remove operation
+                                operations.remove();
+                            }
+                        } else {
+                            operations.remove();
+                        }
```

Comment:
Here for all condition checks, doing the same operations (operations.remove();). Can't we simplify this code block?

Reply:
Yes, of course, we can simplify this first two if conditions, but the main block can not be replaced because it contains different another case that we need to skip by checking. :)